sdk/keyvault/keyvault-admin/src/accessControlClient.ts

```diff
@@ -0,0 +1,459 @@
+    options: ListRoleAssignmentsOptions = {}
+  ): PagedAsyncIterableIterator<KeyVaultRoleAssignment> {
+    const requestOptions = operationOptionsToRequestOptionsBase(options);
+    const span = createSpan("listRoleAssignments", requestOptions);
+    const updatedOptions: ListRoleAssignmentsOptions = {
+      ...requestOptions,
+      ...setParentSpan(span, requestOptions)
+    };
+
+    const iter = this.listRoleAssignmentsAll(roleScope, updatedOptions);
+
+    span.end();
+    return {
+      next() {
+        return iter.next();
+      },
+      [Symbol.asyncIterator]() {
+        return this;
+      },
+      byPage: (settings: PageSettings = {}) =>
+        this.listRoleAssignmentsPage(roleScope, settings, updatedOptions)
+    };
+  }
+
+  /**
+   * @internal
+   * @ignore
+   * Deals with the pagination of {@link listRoleDefinitions}.
+   * @param {string} roleScope The scope of the role definition.
+   * @param {PageSettings} continuationState An object that indicates the position of the paginated request.
+   * @param {ListRoleAssignmentsOptions} [options] Common options for the iterative endpoints.
+   */
+  private async *listRoleDefinitionsPage(
+    roleScope: RoleAssignmentScope,
+    continuationState: PageSettings,
+    options?: ListRoleDefinitionsOptions
+  ): AsyncIterableIterator<KeyVaultRoleDefinition[]> {
+    if (continuationState.continuationToken == null) {
+      const optionsComplete: RoleAssignmentsListForScopeOptionalParams = {
+        // Not supported!
+        // maxresults: continuationState.maxPageSize,
+        ...options
+      };
+      const currentSetResponse = await this.client.roleDefinitions.list(
+        this.vaultUrl,
+        roleScope,
+        optionsComplete
+      );
+      continuationState.continuationToken = currentSetResponse.nextLink;
+      if (currentSetResponse.value) {
+        yield currentSetResponse.value.map(mappings.roleDefinition.generatedToPublic, this);
+      }
+    }
+    while (continuationState.continuationToken) {
+      const currentSetResponse = await this.client.roleDefinitions.listNext(
+        this.vaultUrl,
+        roleScope,
+        continuationState.continuationToken,
+        options
+      );
+      continuationState.continuationToken = currentSetResponse.nextLink;
+      if (currentSetResponse.value) {
+        yield currentSetResponse.value.map(mappings.roleAssignment.generatedToPublic, this);
+      } else {
+        break;
+      }
+    }
+  }
+
+  /**
+   * @internal
+   * @ignore
+   * Deals with the iteration of all the available results of {@link listRoleDefinitions}.
+   * @param {string} roleScope The scope of the role definition.
+   * @param {ListRoleDefinitionsOptions} [options] Common options for the iterative endpoints.
+   */
+  private async *listRoleDefinitionsAll(
+    roleScope: RoleAssignmentScope,
+    options?: ListRoleDefinitionsOptions
+  ): AsyncIterableIterator<KeyVaultRoleDefinition> {
+    const f = {};
```

Comment:
Instead, could you instead make the second param of `listRoleDefinitionsPage` be optional? Passing an empty object in-line would also be preferable.

Reply:
I'll pass {} directly.